CMakeLists.txt

```diff
@@ -268,8 +269,7 @@ include_directories(${PROJECT_SOURCE_DIR}/include)
 # Build the targets
 add_subdirectory(src)
 
-if (BENCHMARK_ENABLE_TESTING)
```

Comment:
BENCHMARK_ENABLE_TESTING != BUILD_TESTING
It's intentionally this way.

Reply:
it is? i thought that we used our own flag only because we didn't have a cmake-level flag. is that not accurate? what's the benefit to having our own flag?

Reply:
> i thought that we used our own flag only because we didn't have a cmake-level flag. is that not accurate?

I wouldn't say so. We can always have any flag we want, it's just the matter of `option(XYZ "" ???)`.

> what's the benefit to having our own flag?

For benchmark itself - none either way.
Nothing in cmake depends on the `BUILD_TESTING` spelling.
There's no benefit from dropping the `BENCHMARK_` prefix.

For consumers, it provides a easier way to toggle the "build testing" per-"subproject".

Reply:
> > i thought that we used our own flag only because we didn't have a cmake-level flag. is that not accurate?
> 
> I wouldn't say so. We can always have any flag we want, it's just the matter of `option(XYZ "" ???)`.
> 
> > what's the benefit to having our own flag?
> 
> For benchmark itself - none either way.
> Nothing in cmake depends on the `BUILD_TESTING` spelling.

except this is automatically created when you include [ctest](https://cmake.org/cmake/help/v3.7/module/CTest.html). if we have both BM_ENABLE_TESTING and BUILD_TESTING then it might be confusing for users who want to disable it.

> There's no benefit from dropping the `BENCHMARK_` prefix.
> 
> For consumers, it provides a easier way to toggle the "build testing" per-"subproject".

true. i'm not sure it's worth the confusion.

Reply:
FWIW i'm not just clinging to stuff just because.
https://github.com/llvm/llvm-project/commit/0addd170ab0880941fa4089c2717f3f3a0e4e25a#diff-b7c61103e5b04a057783630466888503R1024
After this change it will be something like
```
set(zzz BUILD_TESTING)
set(BUILD_TESTING OFF)
...
add_subdirectory(utils/benchmark)

set(BUILD_TESTING zzz)
```
but sure, we've avoided confusion.

Reply:
I was on the "`BUILD_TESTING` team", however, the above argument is pretty solid.

How about a different approach:

No. | `BENCHMARK_BUILD_TESTING` | `BUILD_TESTING` | effect
-- | ------------------------- | --------------- | ------
1 | value                     | no value        | set `BUILD_TESTING` with `BENCHMARK_BUILD_TESTING`
2 | no value                  | value           | keep `BUILD_TESTING`
3 | value                     | value           | overwrite `BUILD_TESTING` with `BENCHMARK_BUILD_TESTING` issue status message if they are different
4 | no value                  | no value        | nothing

After this procedure internally `BUILD_TESTING` is used. `BENCHMARK_BUILD_TESTING` is used only for this procedure.

In 2 there is a risk the `BUILD_TESTING` is "inherited" from the parent project. However, the user can avoid the problem by setting `BENCHMARK_BUILD_TESTING`.

In 3 there are two possible cases. Either user made an inconsistent request (seems less likely) or `BUILD_TESTING` is "inherited" and the user wants to overwrite it with `BENCHMARK_BUILD_TESTING` (seems more likely). Hence, we issue only a status message (if the values are not consistent) since most likely this is expected rather than an error.

Reply:
I'm sorry, i do not understand what problem is being solved by `include(CTest)`.
Why do you think it is needed?

Reply:
It lets us use `ctest` to run the tests in a consistent way across platforms and build systems. (However, maybe I'm mistaken and you don't need `include(CTest)` for that?)

Reply:
> It lets us use `ctest` to run the tests in a consistent way across platforms and build systems. (However, maybe I'm mistaken and you don't need `include(CTest)` for that?)

Yep, `ctest` already just works.

Reply:
Indeed, I checked steps
```bash
git clone https://github.com/google/benchmark.git
git clone https://github.com/google/googletest.git benchmark/googletest
cd benchmark
cmake -E make_directory "build"
cmake -E chdir "build" cmake -DCMAKE_BUILD_TYPE=Release ../
cmake --build "build" --config Release
cmake -E chdir "build" ctest --build-config Release
```
and they just work on current `master`. No need for `include(CTest)`.

I was misled by the statement

> Configure a project for testing with CTest/CDash

at https://cmake.org/cmake/help/v3.5/module/CTest.html.

Reply:
ok, so i'll revert everything but the readme. nice :)

Reply:
Please, make sure that you remove `--parallel` from `README.md` on the line
```bash
$ cmake --build "build" --config Release --parallel
```